File: c++/1137.cpp
//
//  main.cpp
//  test
//
//  Created by 李寻欢 on 2019/10/17.
//  Copyright © 2019 李寻欢. All rights reserved.
//
#include <stdio.h>
#include <iostream>
#include <string>
#include <map>
#include <cmath>
#include <algorithm>
#include <stack>
#include <vector>
#include <queue>
#include <unordered_map>
#include <unordered_set>
#include <string.h>
#include <set>
#include <deque>
using namespace std;
int p,m,n;

struct node{
    string name;
    int online = -1;
    int mid_score = -1;
    int final_score = -1;
    int score = 0;
};

vector<node> v;
unordered_map<string, int> which_idx;

bool cmp(const node &a,const node &b){
    if(a.score != b.score) return a.score > b.score;
    return a.name < b.name;
}

int main(){
   ios::sync_with_stdio(false);
   cin.tie(0);
    cin >> p >> m >> n;
    string name;
    int score;
    int idx = 0;
    for(int i=0;i<p;++i){
        cin >> name >> score;
        if(score < 200) continue;
        which_idx[name] = idx++;
        node cur;
        cur.name = name;
        cur.online = score;
        v.emplace_back(cur);
    }
    for(int i=0;i<m;++i){
        cin >> name >> score;
        if(which_idx.find(name) == which_idx.end()) continue;
        v[which_idx[name]].mid_score = score;
    }
    for(int i=0;i<n;++i){
        cin >> name >> score;
        if(which_idx.find(name) == which_idx.end()) continue;
        v[which_idx[name]].final_score = score;
        if(v[which_idx[name]].mid_score > score)
            v[which_idx[name]].score = int(0.4*v[which_idx[name]].mid_score + 0.6*score+0.5);
        else
            v[which_idx[name]].score = score;
            
    }
        sort(v.begin(), v.end(), cmp);
        for(auto &each:v){
            if(each.final_score < 60) break;
            cout << each.name << " " << each.online <<" " << each.mid_score << " " << each.final_score<<" " << each.score << "\n";
        }
            
}

File: Python3/1137.py
from operator import attrgetter
p,m,n = map(int,input().split())
dic = {}
idx = 0
nodes = []
class node:
    def __init__(self,name,online):
        self.name = name
        self.online = online
        self.mid = -1
        self.final = -1
        self.score = -1

for _ in range(p):
    name,score=  input().split()
    if int(score) < 200:
        continue
    dic[name] = idx
    idx += 1
    nodes.append(node(name,int(score)))

for _ in range(m):
    name,score = input().split()
    if dic.get(name) is None:
        continue
    nodes[dic[name]].mid = int(score)

for _ in range(n):
    name,score = input().split()
    if dic.get(name) is None:
        continue
    score = int(score)
    nodes[dic[name]].final = score
    if nodes[dic[name]].mid > score:
        nodes[dic[name]].score=  -1*int(nodes[dic[name]].mid*0.4+0.6*score+0.5)
    else:
        nodes[dic[name]].score = -score
nodes.sort(key=attrgetter('score','name'))
for node in nodes:
    if -node.score < 60:
        break
    print(node.name,node.online,node.mid,node.final,-node.score)
